src/Microsoft.ApplicationInsights/DataContracts/ExceptionTelemetry.cs

```diff
@@ -293,16 +293,19 @@ public void SerializeData(ISerializationWriter serializationWriter)
         /// <summary>
         /// Set parsedStack from an array of StackFrame objects.
         /// </summary>
-        public void SetParsedStack(System.Diagnostics.StackFrame[] frames)
+        /// <param name="frames">The new array of stack frames.</param>
+        /// <param name="exceptionLevel">Specifies which stack trace should be replaced.
+        /// 0 specifies top level exception, other value selects the respective inner exception.</param>
+        public void SetParsedStack(System.Diagnostics.StackFrame[] frames, int exceptionLevel)
         {
-            if (this.Exceptions != null && this.Exceptions.Count > 0)
+            if (this.Exceptions != null && this.Exceptions.Count > exceptionLevel)
             {
                 if (frames != null && frames.Length > 0)
```

Comment:
Check against `exceptionLevel` instead of 0? I am thinking of using that API and if I get the exception level wrong the code will crash instead of skip.

Reply:
Probably also ignore `exceptionLevel < 0`